src/dask_nested/core.py

```diff
@@ -0,0 +1,390 @@
+        collection = func(graph, *args)
+        return collection
+
+
+class NestedFrame(
+    _Frame, dd.DataFrame
+):  # can use dd.DataFrame instead of dx.DataFrame if the config is set true (default in >=2024.3.0)
+    """An extension for a Dask Dataframe for Nested.
+
+    The underlying non-parallel dataframes are TapeFrames and TapeSeries which extend Pandas frames.
+
+    Examples
+    ----------
+    Instatiation::
+
+        import tape
+        ens = tape.Ensemble()
+        data = {...} # Some data you want tracked by the Ensemble
+        ensemble_frame = tape.EnsembleFrame.from_dict(data, label="my_frame", ensemble=ens)
+    """
+
+    _partition_type = npd.NestedFrame  # Tracks the underlying data type
+
+    def __getitem__(self, key):
+        result = super().__getitem__(key)
+        return result
+
+    @classmethod
+    def from_nested_pandas(
+        cls, data, npartitions=None, chunksize=None, sort=True, label=None, ensemble=None
+    ) -> NestedFrame:
+        """Returns an EnsembleFrame constructed from a TapeFrame.
+
+        Parameters
+        ----------
+        data: `TapeFrame`
+            Frame containing the underlying data fro the EnsembleFram
+        npartitions: `int`, optional
+            The number of partitions of the index to create. Note that depending on
+            the size and index of the dataframe, the output may have fewer
+            partitions than requested.
+        chunksize: `int`, optional
+            Size of the individual chunks of data in non-parallel objects that make up Dask frames.
+        sort: `bool`, optional
+            Whether to sort the frame by a default index.
+        label: `str`, optional
+            The label used to by the Ensemble to identify the frame.
+        ensemble: `tape.Ensemble`, optional
+            A link to the Ensemble object that owns this frame.
+
+        Returns
+        ----------
+        result: `tape.EnsembleFrame`
+            The constructed EnsembleFrame object.
+        """
+        result = dd.from_pandas(data, npartitions=npartitions, chunksize=chunksize, sort=sort)
+        return NestedFrame.from_dask_dataframe(result)
+
+    @classmethod
+    def from_dask_dataframe(cls, df) -> NestedFrame:
+        """Converts a Dask Dataframe to a Dask-Nested NestedFrame
+
+        Parameters
+        ----------
+        df:
+            A Dask Dataframe to convert
+
+        Returns
+        -------
+        `dask_nested.NestedFrame`
+        """
+        return df.map_partitions(npd.NestedFrame)
+
+    def compute(self, **kwargs):
+        """Compute this Dask collection, returning the underlying dataframe or series."""
+        return npd.NestedFrame(super().compute(**kwargs))
+
+    @property
+    def all_columns(self) -> dict:
+        """returns a dictionary of columns for each base/nested dataframe"""
+        all_columns = {"base": self.columns}
+        for column in self.columns:
```

Comment:
Minor question but how does this relate to `compute` in dask land?

If I drop columns but have not yet computed, what should we return?

Reply:
This just returns the current columns of the NestedFrame, so it works exactly as dask's columns property in the sense that it will track changes to these columns in the case of things like column dropping without the need to compute

Reply:
Makes sense!